zebra-chain/src/utils.rs

```diff
@@ -0,0 +1,15 @@
+//! Utility functions to be used in several crates.
+
+/// Returns the hexadecimal-encoded string `s` in byte-reversed order.
+pub fn byte_reverse_hex(s: &str) -> String {
```

Comment:
This is fine but normally I'd prefer to use the `Display` trait to handle reversing the representation rather than constructing a new reversed string.

Reply:
@yaahc how would that work?

Zebra uses a consistent byte order internally, so we should never print out byte reversed strings from our own data structures.

But we need to byte-reverse strings from zcashd and similar tools, because their outputs are in the opposite byte order.

Currently, the two use cases are:
* the zebrad revhex command, to manually reverse a string
* reversing strings from zcash-cli during checkpoint generation

Reply:
If we're reading in a single BlockHeaderHash from zcashd, we should do the reversal during parsing.

Reply:
(And if we're doing the reversal during parsing, then we can move this code to the serialisation module, and delete the utils module.)

Reply:
@teor2345  the way this would work would be to add a struct that wraps a `&str` and implements `Display` which does most of the logic in this fn except instead of collecting things back into a string it repeatedly calls `f.write_str(..)`

Then we could implement an extension trait that just wraps the input in this type and implement this trait for `&str` so then you can just call `println!("{}", input.trim().rev_hex());` to get the reversed hex version of the input string without allocating any additional strings.